### PR TITLE
Add ecs compatible versions of base-focal and ruby-focal

### PR DESCRIPTION
**Pull request by jsondheimer**
These versions will not set the ARK_HOST or ARK_HOSTNAME variables.

### CODE REVIEW

File: base-focal-ecs/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+RUN gem2.7 install --no-document aws-sdk-resources --pre -v '>= 2.11.562'
+
+COPY ship /bin/
+COPY ship.d /etc/ship.d/
+
+# Add shared files from shipyard repo, this is done to keep these files in sync across all images
+ADD https://raw.githubusercontent.com/socrata/shipyard/main/files/env_parse /bin/
+ADD https://raw.githubusercontent.com/socrata/shipyard/main/files/set_local_dev_hostname /bin/
+# Credential management bits
+ADD https://raw.githubusercontent.com/socrata/shipyard/main/files/clortho-get /etc/pre-init.d/clortho-get
```
Comment (sarah-salinas) [inline base-focal-ecs/Dockerfile:43]: You can remove this line for ECS

File: base-focal-ecs/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#Create Data Directory
+ENTRYPOINT ["/bin/ship"]
+CMD ["run"]
+
+# Ensure that containers and apps default to UTF-8
+RUN locale-gen en_US.UTF-8
+ENV LANG en_US.UTF-8
+
+# LABEL must be last for proper base image discoverability
+LABEL repository.socrata/base-focal=""
```
Comment (sarah-salinas) [inline base-focal-ecs/Dockerfile:58]: Shouldn't this be `base-focal-ecs`?

File: ruby-focal/3.2.x-ecs/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+RUN mkdir -p /usr/local/etc \
+	&& { \
+		echo 'install: --no-document'; \
+		echo 'update: --no-document'; \
+	} >> /usr/local/etc/gemrc
+
+ENV RUBY_MAJOR 3.2
+
+# some of ruby's build scripts are written in ruby
+#   we purge system ruby later to make sure our final image uses what we just built
```
Comment (sarah-salinas) [inline ruby-focal/3.2.x-ecs/Dockerfile:14]: Lies . . . but I know this is copied and pasted.

File: base-focal-ecs/Dockerfile (hunk 2 above)
Comment (jsondheimer) [reply base-focal-ecs/Dockerfile]: base-focal:ecs actually, but yes

File: ruby-focal/3.2.x-ecs/Dockerfile (hunk 3 above)
Comment (jsondheimer) [reply ruby-focal/3.2.x-ecs/Dockerfile]: Might as well start telling the truth now.

**Comment by sarah-salinas** [comment]
I know I recommended removing the clortho file . . . but as I was thinking about this . . . we want the base image to be able to run in marathon and ECS, not just ECS.  That means keeping the clortho file -- it's easily bypassed already.  And adding in the set_ark_host file, but making all of it conditional on the pre-existence of the ARK_HOST environment variable.

**Comment by jsondheimer** [comment]
Oh, I thought we were making a separate base image to use only in ECS, something to transition to as we move off of mesos. 

This also really just needs to be a bandaid for the things that have to be run in ECS for now until I get the new noble base images built in ECR.

**Comment by sarah-salinas** [comment]
In this case, we plan to have frontend running in marathon and frontend browse running in ECS, so we need it to work in  both, or we'll have to do more work.
The longer term goal is to transition services completely over, and in that case, we can fully remove the ARK HOST and CLORTHO pieces.

**Comment by jsondheimer** [comment]
Ok, so the ecs tagged ones should be conditional like discussed earlier? But having them tagged separately at least limits the scope in case the changes break unexpectedly?

**Comment by sarah-salinas** [comment]
Yes, having them tagged separately allows us to choose which services will use them so we can test that they are working properly one at a time.

**Comment by jsondheimer** [comment]
ARK_HOST and ARK_HOSTNAME will be set if AWS_EXECUTION_ENV is unset
```
$ docker run -e AWS_EXECUTION_ENV socrata/ruby-focal:3.2.x-ecs env 2> /dev/null
DEBUG: AWS_REGION unset. Defaulting to us-west-2
INFO: Either CLORTHO_BUCKET or CLORTHO_PATH unset. Skipping creds download
no_proxy=localhost,127.0.0.1,169.254.169.254,jenkins
HOSTNAME=8f14fcd62737
ARK_HOSTNAME=ark-host
ARK_HOST=172.17.0.1
PWD=/
RUBY_MAJOR=3.2
HOME=/root
LANG=en_US.UTF-8
SHLVL=0
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
DEBIAN_FRONTEND=noninteractive
```

ARK_HOST and ARK_HOSTNAME will _not_ be set if AWS_EXECUTION_ENV is "AWS_ECS_FARGATE"
```
$ docker run -e AWS_EXECUTION_ENV=AWS_ECS_FARGATE socrata/ruby-focal:3.2.x-ecs env 2> /dev/null
DEBUG: AWS_REGION unset. Defaulting to us-west-2
INFO: Either CLORTHO_BUCKET or CLORTHO_PATH unset. Skipping creds download
no_proxy=localhost,127.0.0.1,169.254.169.254,jenkins
AWS_EXECUTION_ENV=AWS_ECS_FARGATE
HOSTNAME=9612378e5182
PWD=/
RUBY_MAJOR=3.2
HOME=/root
LANG=en_US.UTF-8
SHLVL=0
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
DEBIAN_FRONTEND=noninteractive
```

**Review by sarah-salinas — APPROVED** [review]
Thank you for reworking this to conditionally include ark host.  I think the conditional you chose is the most specific for what we're trying to solve.  Everything else looks good to me.

**Pull request merged**